eclair-core/src/main/scala/fr/acinq/eclair/io/ReconnectionTask.scala

```diff
@@ -121,6 +121,10 @@ class ReconnectionTask(nodeParams: NodeParams, remoteNodeId: PublicKey) extends
   }
 
   whenUnhandled {
+    case Event("connected", _) => stay
+
+    case Event(Status.Failure(_: Client.ConnectionFailed), _) => stay
```

Comment:
In which case do we receive this event outside of the `CONNECTING` state?

Reply:
If there are two concurrent connections (one inbound the other outbound), the inbound connection works and the outbound connection fails. Happens all the time if a peer doesn't maintain an accurate public address e.g.